microsoft/testsuites/vm_extensions/runtime_extensions/vmaccess.py

```diff
@@ -0,0 +1,283 @@
+    TestSuiteMetadata,
+    simple_requirement,
+)
+from lisa.operating_system import BSD
+from lisa.sut_orchestrator import AZURE
+from lisa.sut_orchestrator.azure.features import AzureExtension
+from microsoft.testsuites.vm_extensions.runtime_extensions.common import (
+    create_and_verify_vmaccess_extension_run,
+)
+
+
+def _validate_password(
+    node: Node, username: str, password: str, valid: bool = True
+) -> None:
+    message = f"Password not set as intended for user {username}."
+    # simple command to determine if username password combination is valid/invalid
+    node.execute(
+        cmd=f'echo "{password}" | su --command true - {username}',
+        shell=True,
+        expected_exit_code=0 if valid else 1,
+        expected_exit_code_failure_message=message,
+    )
+
+
+def _validate_ssh_key_exists(node: Node, username: str, exists: bool = True) -> None:
+    # Command checks whether authorized_keys file is created and has correct format
+    message = f"Public key file for user {username} \
+        {'does not exist' if exists else 'exists'}."
+    node.execute(
+        cmd=f"ssh-keygen -l -f /home/{username}/.ssh/authorized_keys",
+        shell=True,
+        sudo=True,
+        expected_exit_code=0 if exists else 255,
+        expected_exit_code_failure_message=message,
+    )
+
+
+def _validate_account_expiration_date(
+    node: Node, username: str, expiration_str: str
+) -> None:
+    message = f"Cannot retrieve account details for user {username}"
+    # Command checks whether the account expiration specified is correct
+    result = node.execute(
+        cmd=f"chage -l {username}",
+        shell=True,
+        sudo=True,
+        expected_exit_code=0,
+        expected_exit_code_failure_message=message,
+    )
+
+    assert_that(result.stdout).described_as(
+        f"Expected the account details to contain expiration date of {expiration_str}"
+    ).contains(expiration_str)
+
+
+@TestSuiteMetadata(
+    area="vm_extension",
+    category="functional",
+    description="""
+    This test suite tests the functionality of the VMAccess VM extension.
+
+    Settings are protected unless otherwise mentioned.
+    OpenSSH format public keys correspond to ssh-rsa keys.
+
+    It has 8 test cases to verify if VMAccess runs successfully when provided:
+        1. Username and password
+        2. Username and OpenSSH format public key
+        3. Username with both a password and OpenSSH format public key
+        4. Username with no password or ssh key (should fail)
+        5. Username and certificate containing public ssh key in pem format
+        6. Username and SSH2 format public key
+        7. Username to remove
+        8. Username, OpenSSH format public key, and valid expiration date
+    """,
+    requirement=simple_requirement(
+        supported_features=[AzureExtension],
+        supported_platform_type=[AZURE],
+        unsupported_os=[BSD],
+    ),
+)
+class VMAccessTests(TestSuite):
+    _OPENSSH_KEY = (
```

Comment:
@LiliDeng please give the sample code to generate ssh keys, instead of hard code them here. It won't pass security checks.

Reply:
try this
```
import paramiko

# Create an instance of the RSAKey class
key = paramiko.RSAKey.generate(2048)

# Save the private key to a file
key.write_private_key_file('id_rsa', 'ssh2')
#key.write_private_key_file('id_rsa', 'openssh')
#key.write_private_key_file('id_rsa', 'pem')

# Save the public key to a file
with open('id_rsa.pub', 'w') as f:
    f.write(f'{key.get_name()} {key.get_base64()}')
```

Reply:
Generating the keys from VM is a way. You can find a command line tool to do that, and add the tool into LISA.